db/db.go

```diff
@@ -68,19 +64,58 @@ type DB struct {
 
 // NewDB creates a new DB, which will *own* ds and dispatcher for internal use.
 // Saying it differently, ds and dispatcher shouldn't be used externally.
-func NewDB(ts service.Service, opts ...Option) (*DB, error) {
+func NewDB(ctx context.Context, ts s.Service, id thread.ID, opts ...Option) (*DB, error) {
 	config := &Config{}
 	for _, opt := range opts {
 		if err := opt(config); err != nil {
 			return nil, err
 		}
 	}
-	return newDB(ts, config)
+
+	ti, err := ts.GetThread(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if len(ti.Logs) == 0 {
+		ti, err = ts.CreateThread(ctx, id, s.FollowKey(sym.New()), s.ReadKey(sym.New()))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return newDB(ts, id, config)
+}
+
+// NewDBFromAddr creates a new DB from a thread hosted by another peer at address,
+// which will *own* ds and dispatcher for internal use.
+// Saying it differently, ds and dispatcher shouldn't be used externally.
+func NewDBFromAddr(ctx context.Context, ts s.Service, addr ma.Multiaddr, followKey, readKey *sym.Key, opts ...Option) (*DB, error) {
```

Comment:
I see that in `newDb` the _DB_ is started.
Shouldn't this method also receive the Models to register before the start, to avoid the problem of receiving patches without having the models to validate? (If that's the case (because i'm maybe missing something), would be nice to add that to the comments?)

Reply:
Sure, we could allow collections to be created here. I didn't do that originally because as long as we allow them to be added _at any point_, including after the db is started, the safety measure of adding them when the db is started is sorta mute, right?

Reply:
Say I want to join an existing _Db_ in the safest way possible.
What would be the correct flow?

Asking that because if the first line of code is `NewDBFromAddr`, then that isn't safe since it will start syncing existing patches possibly without giving me time to register the necessary models (even if the next line in the code is registering the models). Am I missing some other existing _Db_ bootstrap code flow?

Reply:
Nope, you're correct... but the solution is a bit more involved. We need a way to load a the models from the address. Otherwise, we're back in the same position as before: I have to manually start the DB.

Reply:
Well, the long-term solution I completely agree should be some ideas we brainstormed some days ago about models being first-class citizens.

But, if you can receive a slice of Models as a param in this method which the method registers before calling the `Start`, should suffice to avoid this problem and still auto-starts the _DB_.

Just asking if it's a timing decision since the full solution will be short-term, or maybe this idea was missed.

Reply:
To be clear, just mentioning what could be a safer temporal solution to that problem. That doesn't change the fact that a less safe one might be best to avoid changing the API again later.

Reply:
I see what you mean! Okay, I'll add the register models to this method. Do you agree the same pattern is not needed in `NewDB`?

Reply:
Yep, `NewDB` doesn't need that since nothing should happen between that `NewDB` and the next line that some models are registered!

Reply:
Okay, sounds good. Agreed. Change coming after I finish updating the Android client.